elyra/pipeline/pipeline.py

```diff
@@ -308,6 +355,14 @@ def memory(self) -> Optional[str]:
     def gpu(self) -> Optional[str]:
         return self._component_params.get("gpu")
 
+    @property
+    def mounted_volumes(self) -> List[VolumeMount]:
+        return self._component_params.get(MOUNTED_VOLUMES)
```

Comment:
What is preventing `GenericOperations` from using the same location for `mounted_volumes` and `kubernetes_secrets` as `Operations` (i.e., in `app_data` rather than `app_data.component_parameters`) or vice versa?  This difference is referenced in the opening description and it seems like it would be worth a possible migration to place these pieces of information in the same location.

Using a property to handle this difference is nice, but it would be good if we could have as few differences within the file structure as we can.

Reply:
Hmm yeah not sure why I hadn't thought of that as an option. Thinking through this, it looks like the main thing that would have to be worked through is that this would necessitate different logic among those properties that need to be propagated. For example, we need to propagate both `env_vars` and `mounted_volumes` (and presumably `env_vars` should stay in the `app_data.component_parameters` stanza). That being said, once we got past that part, this approach would probably result in fewer differences between `Operation`s and `GenericOperation`s. I'm going to give it a try and hope that nothing weird comes up 🙂